### PR TITLE
Allow coercion between struct types

### PR DESCRIPTION
**Pull request by jdidion**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Valid examples WDL's were added or updated to the SPEC.md (see the [guide](https://github.com/openwdl/wdl-tests/blob/main/docs/MarkdownTests.md) on writing markdown tests)

### CODE REVIEW

**Review by markjschreiber — APPROVED** [review]
LGTM

File: SPEC.md [hunk 1]
```diff
@@ -1701,7 +1703,78 @@ Example output:
 - If a `Struct` (or `Object`) declaration is initialized using the struct-literal (or object-literal) syntax `Words literal_syntax = Words { a: ...` then the keys will be `"a"`, `"b"` and `"c"`.
 - If a `Struct` (or `Object`) declaration is initialized using the map-literal syntax `Words map_coercion = { a: ...` then the keys are expressions, and thus `a` will be a variable reference to the previously defined `String a = "beware"`.
 
-###### 🗑 Limited exceptions
+###### Struct-to-Struct Coercion
+
+Two `Struct` types are considered compatible when the following are true:
+
+1. They have the same numbers of members
```
Comment (illusional) [inline SPEC.md]:
```suggestion
1. They have the same number of members
```

**Review by illusional — APPROVED** [review]
Do we have any way we handle bullet list punctuation, like commas at the end of lines, or full stops at the end of the last element? Pedantic from me, but thought I'd ask.

File: SPEC.md [hunk 2]
```diff
@@ -1710,7 +1712,78 @@ Example output:
 - If a `Struct` (or `Object`) declaration is initialized using the struct-literal (or object-literal) syntax `Words literal_syntax = Words { a: ...` then the keys will be `"a"`, `"b"` and `"c"`.
 - If a `Struct` (or `Object`) declaration is initialized using the map-literal syntax `Words map_coercion = { a: ...` then the keys are expressions, and thus `a` will be a variable reference to the previously defined `String a = "beware"`.
 
-###### 🗑 Limited exceptions
+###### Struct-to-Struct Coercion
+
+Two `Struct` types are considered compatible when the following are true:
+
+1. They have the same number of members
```
Comment (jdidion) [inline SPEC.md]:
```suggestion
1. They have the same number of members.
```

File: SPEC.md [hunk 3]
```diff
@@ -1710,7 +1712,78 @@ Example output:
...
+2. Their members' names are identical
```
Comment (jdidion) [inline SPEC.md]:
```suggestion
2. Their members' names are identical.
```

File: SPEC.md [hunk 4]
```diff
@@ -1710,7 +1712,78 @@ Example output:
...
+3. The type of each member in the source struct is coercible to the type of the member with the same name in the target struct
```
Comment (jdidion) [inline SPEC.md]:
```suggestion
3. The type of each member in the source struct is coercible to the type of the member with the same name in the target struct.
```

**Review by jdidion — COMMENTED** [review]
It looks like elsewhere bullets end in period

**Pull request merged**